### PR TITLE
starting PR to add auth for marathon

### PR DESCRIPTION
**Pull request by anshupande**
I will work with @fraczles  to implement the same in FD as well. Also I need to figure out how to pass the username and password variables

### CODE REVIEW

**Comment by anshupande** [comment]
In FD we can access API as follows: curl -u ${MARATHON_USER}:${MARATHON_PASSWORD} ... @fraczles

File: v2/fleet-local/control/marathon@.service [hunk 1]
```diff
@@ -21,10 +21,14 @@ ExecStartPre=-/usr/bin/docker rm marathon
 ExecStart=/usr/bin/sh -c "/usr/bin/docker run \
   --name marathon \
   -e LIBPROCESS_PORT=9090 \
+  -e marathon_user=`etcdctl get /marathon_user` \
+  -e marathon_password=`/etcdctl get marathon_password` \
```
Comment (fraczles) [inline v2/fleet-local/control/marathon@.service]: might be a good idea to scope these, i.e
``-e MARATHON_USER=`etcdctl get /FD/MARATHON_USER` \``

File: v2/fleet-local/control/marathon@.service [hunk 2]
```diff
@@ -21,10 +21,14 @@ ExecStartPre=-/usr/bin/docker rm marathon
 ExecStart=/usr/bin/sh -c "/usr/bin/docker run \
   --name marathon \
   -e LIBPROCESS_PORT=9090 \
+  -e marathon_user=`etcdctl get /marathon_user` \
```
Comment (fraczles) [inline v2/fleet-local/control/marathon@.service]: `-e MARATHON_USER=`etcdctl get /marathon_user` \`
`-e MARATHON_PASSWORD=`etcdctl get /marathon_password` \`

File: v2/setup/marathon.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+source /etc/environment
+
+HOMEDIR=$(eval echo "~`whoami`")
+
+sudo docker run --rm \
+    -v ${HOMEDIR}:/data/ behance/docker-aws-s3-downloader \
+     us-east-1 $CONTROL_TIER_S3SECURE_BUCKET .marathon
+
+SYSDIG_KEY=$(sudo cat ${HOMEDIR}/.marathon)
```
Comment (fraczles) [inline v2/setup/marathon.sh]: `SYSDIG_KEY`?

File: v2/setup/marathon.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+etcdctl set /marathon_user $marathon_user
+etcdctl set /marathon_password $marathon_password
```
Comment (fraczles) [inline v2/setup/marathon.sh]: how are `$marathon_user` and `$marathon_password` being set?
I don't know bash too well.

Comment (anshupande) [inline v2/setup/marathon.sh]: How did we set this etcdctl set /FD/FD_DB_USERNAME $FLIGHT_DIRECTOR_DB_USERNAME we will use similar approach, I am also not an expert here... but are we not passing this from secure S3 bucket?

**Comment by anshupande** [comment]
@chr0n1x please review this once you get time!

File: v2/fleet-local/control/marathon@.service [hunk 5]
```diff
@@ -21,10 +21,13 @@ ExecStartPre=-/usr/bin/docker rm marathon
 ExecStart=/usr/bin/sh -c "/usr/bin/docker run \
   --name marathon \
   -e LIBPROCESS_PORT=9090 \
+  -e MARATHON_USER=`etcdctl get /marathon_user` \
```
Comment (chr0n1x) [inline v2/fleet-local/control/marathon@.service]: `/marathon/username` & `/marathon/password`

File: v2/setup/marathon.sh [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+source /etc/environment
+
+HOMEDIR=$(eval echo "~`whoami`")
+
+sudo docker run --rm \
+    -v ${HOMEDIR}:/data/ behance/docker-aws-s3-downloader \
+     us-east-1 $CONTROL_TIER_S3SECURE_BUCKET .marathon
+
+
+etcdctl set /marathon_user $MARATHON_USER
```
Comment (chr0n1x) [inline v2/setup/marathon.sh]: `/marathon/username` & `/marathon/password`

Comment (chr0n1x) [inline v2/setup/marathon.sh]: Also, I would assume that you're reading these from the `.marathon` file above.
You might want to actually read the file. It's not going to be in the environment.

File: v2/fleet-local/control/marathon@.service (hunk 5 above)
Comment (chr0n1x) [inline v2/fleet-local/control/marathon@.service]: these two lines aren't needed.

File: v2/fleet-local/control/marathon@.service [hunk 7]
```diff
@@ -21,10 +21,13 @@ ExecStartPre=-/usr/bin/docker rm marathon
 ExecStart=/usr/bin/sh -c "/usr/bin/docker run \
```
Comment (chr0n1x) [inline v2/fleet-local/control/marathon@.service:23]: you're probably going to need something more along the lines of:

`Environment="USER=etcdctl get /marathon/username"`

And in the command below **for the container** `docker run ... $($IMAGE) --http_credentials $($USER):$($PW)`

Comment (anshupande) [inline v2/fleet-local/control/marathon@.service:23]: Updated the same, plz review again.... How can we update the $CONTROL_TIER_S3SECURE_BUCKET with credentials now? @chr0n1x

**Pull request merged**